avm/ptn/authorization/resource-role-assignment/main.bicep

```diff
@@ -0,0 +1,106 @@
+metadata name = 'ResourceRole Assignments (All scopes)'
+metadata description = 'This module deploys a Role Assignment for a specific resource.'
+metadata owner = 'Azure/module-maintainers'
+
+@sys.description('Required. The scope for the role assignment, fully qualified resourceId.')
+param resourceId string
+
+@sys.description('Required. The unique guid name for the role assignment.')
+param name string
```

Comment:
```suggestion
@sys.description('Optional. The unique guid name for the role assignment.')
param name string?
```

Reply:
Note, for this and all other changes, I'd recommend to re-run `Set-AvmModule -ModuleFolderPath '.\avm\ptn\authorization\resource-role-assignment' -SkipFileAndFolderSetup -ThrottleLimit 5 -Recurse`

Reply:
> Note, for this and all other changes, I'd recommend to re-run `Set-AvmModule -ModuleFolderPath '.\avm\ptn\authorization\resource-role-assignment' -SkipFileAndFolderSetup -ThrottleLimit 5 -Recurse`

just bubbling this up in its own comment for awareness

Reply:
I have a very strange consequence of making the `name` optional. Once I make it optional, the bicep linter gives an error later down the file:
`
The symbol "tFile" is declared in the context of the outer deployment and cannot be accessed by expressions within a nested deployment template that uses inner scoping for expression evaluation.
`

If I remove the optional `?` decorator from name, it works again.

Any suggestion?

Reply:
Should I make it required with a default value of empty string `''`?

Reply:
Hey @peterbud, sorry for the late reply (I'm actually out of office), but I'll see to clone your repository to try and replicate the behavior. The 2 subjects seem not to overlap. I'll let you know once I get around to it.

Reply:
Hey @peterbud,
I was able to reproduce the issue (which does not make any sense to me and may be a language bug). In any case, I did use your fork to implement an update to the Pester tests that you could find here: https://github.com/Azure/bicep-registry-modules/pull/1917

If it's merged sooner than later, I'd suggest to go with the original plan, which is
- Make the `name` variable nullable
- Remove the introduced explicit variable and invoke the function where the value is used directly

I'll ask the team to please fasttrack the review of the linked PR.

Reply:
Thanks, good news! Once your PR is merged, I'll move ahead with the suggested solution.

Reply:
Hi @AlexanderSehr , the updated Pester tests are working fine, and there is no test case error now with `LoadJsonContent`. 

However making the name optional and adding a handling of null `name` in the template assignment with formulas failed at deployment with a strange error message:

`Status Message: Deployment template validation failed: 'Template language version '2.0' does not support 'Outer' expression evaluation. Ensure you have set expressionEvaluationOptions.scope to 'Inner'.'. (Code:InvalidTemplate)`

More [here](https://github.com/peterbud/bicep-registry-modules/actions/runs/9066384300/job/24909256879#step:4:551)

I have tried different approaches, at the end which was working was to assign a default value to the `name` param - see the last commit. This way this param is still optional. Let me know if that's OK, or you have a better alternative.